src/no-side-effects.spec.ts

```diff
@@ -0,0 +1,161 @@
+                return prefix;
+              }`,
+        filename: 'src/no-side-effects.ts',
+      },
+      {
+        code: `import { strict as assert } from 'node:assert';
+               import debug from 'debug';
+               import Koa from 'koa';
+               import { v4 as uuid } from 'uuid';
+               import Router from '@koa/router';
+               import { StatusCodes } from 'http-status-codes';    
+               
+               const log = debug('report:event');
+              
+               const reportName = 'TEST_REPORT';
+
+               type records = Record;
+               
+               interface Report {
+                  reportName: string;
+                  report: [];
+               }
+               
+               function createS3ObjectKey(
+                from: string,
+                to: string,
+                bin: string,
+                reportId: string,
+                createdAt = new Date().toISOString(),
+              ): string {
+                // YYYY/MM/DD/HH
+                const prefix = formatUtc(createdAt, 'yyyy/MM/dd/HH');
+                return prefix;
+              }
+              
+              const jsonSchemaValidator = new Ajv({ allErrors: true }).compile(schema);`,
+        filename: 'src/side-effects.spec.ts',
+      },
+    ],
+    invalid: [
+      {
+        code: `import { strict as assert } from 'node:assert';
+               import debug from 'debug';
+               import Koa from 'koa';
+               import { v4 as uuid } from 'uuid';
+               import Router from '@koa/router';
+               import { StatusCodes } from 'http-status-codes';    
+               
+               const log = debug('report:event');
+              
+               const reportName = 'TEST_REPORT';
+
+               type records = Record;
+               
+               interface Report {
+                  reportName: string;
+                  report: [];
+               }
+               
+               function createS3ObjectKey(
+                from: string,
+                to: string,
+                bin: string,
+                reportId: string,
+                createdAt = new Date().toISOString(),
+              ): string {
+                // YYYY/MM/DD/HH
+                const prefix = formatUtc(createdAt, 'yyyy/MM/dd/HH');
+                return prefix;
+              }
+              assert(\`I'm a number, \${numberValue}\`);
+              const jsonSchemaValidator = new Ajv({ allErrors: true }).compile(schema);`,
+        errors: [{ message: 'No side effects can occur at the main module-level' }],
+        filename: 'src/side-effects.ts',
+      },
+      {
+        code: `import { strict as assert } from 'node:assert';
+                import http from 'node:http';
+                import debug from 'debug';
+                
+                import { getConfiguration, logger, root, service } from '@checkdigit/serve-runtime';
```

Comment:
this is a public repo, we shouldn't have examples that make use of internal/private modules